subgraphs/vesta-finance/src/entities/market.ts

```diff
@@ -0,0 +1,375 @@
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import {
+  InterestRate,
+  Market,
+  MarketDailySnapshot,
+  MarketHourlySnapshot,
+} from "../../generated/schema";
+import {
+  ACTIVE_POOL_CREATED_BLOCK,
+  ACTIVE_POOL_CREATED_TIMESTAMP,
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  INT_ZERO,
+  LIQUIDATION_FEE_PERCENT,
+  MAXIMUM_LTV,
+  SECONDS_PER_DAY,
+  SECONDS_PER_HOUR,
+} from "../utils/constants";
+import {
+  addProtocolMarketAssets,
+  addProtocolBorrowVolume,
+  addProtocolDepositVolume,
+  addProtocolLiquidateVolume,
+  decrementProtocolOpenPositionCount,
+  getOrCreateFinancialsSnapshot,
+  getOrCreateLendingProtocol,
+  incrementProtocolPositionCount,
+  updateProtocolBorrowBalance,
+  updateProtocolUSDLocked,
+  addProtocolWithdrawVolume,
+  addProtocolRepayVolume,
+} from "./protocol";
+import { getOrCreateAssetToken, getCurrentAssetPrice } from "./token";
+import { bigIntToBigDecimal, exponentToBigDecimal } from "../utils/numbers";
+import { getOrCreateStableBorrowerInterestRate } from "./rate";
+import { EventType } from "./event";
+
+export function getOrCreateMarket(asset: Address): Market {
+  const id = asset.toHexString();
+  let market = Market.load(id);
+  if (!market) {
+    const id = asset.toHexString();
+    const inputToken = getOrCreateAssetToken(asset);
+    market = new Market(id);
+    market.protocol = getOrCreateLendingProtocol().id;
+    market.name = inputToken.name;
+    market.isActive = true;
+    market.canUseAsCollateral = true;
+    market.canBorrowFrom = true;
+    market.maximumLTV = MAXIMUM_LTV;
```

Comment:
is there a way to dynamically get this in vesta? seems to be different by different vaults on their frontend

Reply:
Fixed at d85b20c7bda0a79872d3a08825b6b0e7fb7e3ba3 Thanks!